### ISSUE TITLE
About the importance metric Z

### ISSUE THREAD

**Issue by sichun233746** [JHLew/SoftSplat-Full#1]
Hello,

Thanks for the well implementation of Softmax Splatting.

I'm curious about the importance matric part of your code:
https://github.com/JHLew/SoftSplat-Full/blob/0c0e6640b09db6aab3c4ace4cabbc86fa0e3b840/SoftSplatModel.py#L93
I think you should substract the input frames with the back warped results.
Or did I miss anything in your code?

Also, can I get the PSNR value of 35.64 by directly training with your implementation?

Really thank you for your implementation.

**Comment by JHLew** [comment]
Thank you for your interest, and you are absolutely correct.

It is in fact a huge mistake of mine.
Thank you for pointing it out, and I will fix and update the related materials (code & pretrained weights) accordingly in a couple of days.

To be honest, I am a bit surprised that my current implementation worked out well.

Still, the pretrained weights and implementation I have shared at the moment did lead to PSNR of 35.64 in Vimeo90k.

I anticipate that it will be improved once I have dealt with the issue.
Thank you again for pointing it out, and I will let you know once everything has been updated!

**Comment by sichun233746** [comment]
Thanks for quick response.

The good performance let me be curious about do the Z map did affect the performance a lot.

But still thanks for sharing.

**Comment by JHLew** [comment]
Hi, I have retrained the model and shared the pre-trained model.

The PSNR of the model is 35.59, which is slightly below 35.64, but I don't think there is a much of a crucial difference.
(may be due to different random seeds, etc. It may be improved with better seeds / training scheduling, but I am not willing to put more effort in reaching a better performance...)

Just wanted to let you know just in case!

According to this experiment, I too have come to wonder on how the Z affects the performance, as you have mentioned above......

**Comment by sichun233746** [comment]
Thanks for providing everything.

FYI, the author of sotfmax splatting has recently publisedh a related paper.
You may want to check it.

**Comment by JHLew** [comment]
Yes, I am aware of it and currently trying to reproduce it, too.
Thanks!

**Comment by sichun233746** [comment]
Hi,

I further investigate your pre-train weight and find one issue:

Your alpha is about -0.15, which will lead to similar situation with average splatting.
You can try to load your pre-train weight and use "average" splatting,
then you will get a similar PSNR number around 35.59.

I think the proper alpha value is around -20.

**Comment by JHLew** [comment]
Hi, sorry for the late response.

I got back to this again recently, and tried several more experiments along with some updates.
I was aware that the alpha value is to be around -20 ideally, but did not pay much attention to it before you mentioned it.
I tried investigating more, but the alpha value did not seem to reach a value around -20, but somewhere around -0.2.
Initializing the alpha with values such as -20 or -10 did not help, but rather the training was unstable and eventually collapsed.

The official implementation of the full model is unavailable, so it is hard to tell if this was also the case in the original model,
or it could be just the issue of my implementation (with a higher possibility). Some minor details of the implementation are missing from the paper, so the difference may have rooted from those details... Although I think it would still be "minor"...

I would be glad to hear about anything that you have figured according to this issue!
